protocols/relay/examples/relay.rs

```diff
@@ -18,14 +18,41 @@
 // FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 // DEALINGS IN THE SOFTWARE.
 
+//! A basic relay server and relay client implementation.
+//!
+//! The example below involves three nodes: (1) a relay server, (2) a listening
+//! relay client listening via the relay server and (3) a dialing relay client
+//! dialing the listening relay client via the relay server.
+//!
+//! 1. To start the relay server, run `cargo run --example relay -- relay`.
+//!
+//! 2. To start the listening relay client run `cargo run --example relay -- client-listen
+//! <addr-relay-server>/p2p/<peer-id-relay-server>/p2p-circuit` in a second terminal where:
+//!
+//!   - `<addr-relay-server>` is replaced by one of the listening addresses of the relay server.
+//!   - `<peer-id-relay-server>` is replaced by the peer id of the relay server.
+//!
+//! 3. To start the dialing relay client run `cargo run --example relay -- client-listen
+//! <addr-relay-server>/p2p/<peer-id-relay-server>/p2p-circuit/<peer-id-listening-relay-client>` in
```

Comment:
suppose that the relay peer id is: QmRelay and the relay peer address is /ip6/::1/tcp/54443
the listening peer id is:  QmListenPeer

cargo run --example relay -- client-dial /ip6/::1/tcp/54443/p2p/QmRelay/p2p-circuit/p2p/QmListenPeer